### PR TITLE
pytest.main: collect: factor out _visit_filter

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

File: src/_pytest/main.py [hunk 1]
```diff
@@ -636,6 +626,18 @@ def _recurse(self, dirpath):
         ihook.pytest_collect_directory(path=dirpath, parent=self)
         return True
 
+    if six.PY2:
+
+        @staticmethod
+        def _visit_filter(f):
```
Comment (RonnyPfannschmidt) [inline src/_pytest/main.py]: i recall this can be stored as constant using `methodcaller('check', file=True, notext="pyc")`

Comment (blueyed) [reply src/_pytest/main.py]: Not sure if I understand correctly, but there is no "notext" in py.

Comment (RonnyPfannschmidt) [reply src/_pytest/main.py]: there is a ext checker method and the not prefix of the kwarg triggers inversion ^^
its horrendous in a funny way

Comment (blueyed) [reply src/_pytest/main.py]: Would rather keep it like it is now (i.e. only move it), but feel free to push an update, or patch here.

Comment (blueyed) [reply src/_pytest/main.py]: @RonnyPfannschmidt 
What do you plan?  Either post a patch/push a fixup (feel free to squash), or approve it, please.

Comment (RonnyPfannschmidt) [reply src/_pytest/main.py]: yikes, thanks for the call out, i missed that bit

**Pull request merged**